tests/phpunit/tests/compat/str_ends_with.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * @group compat
+ *
+ * @covers ::str_ends_with
+ */
+class Tests_Compat_str_ends_with extends WP_UnitTestCase {
+
+	/**
+	 * Test that is_iterable() is always available (either from PHP or WP).
+	 *
+	 * @ticket 43619
+	 */
+	public function test_is_iterable_availability() {
+		$this->assertTrue( function_exists( 'str_ends_with' ) );
+	}
+	/**
+	 * @dataProvider str_ends_with_provider
+	 */
+	public function test_str_ends_with( $expected, $string, $needle ) {
+		if ( ! function_exists( 'str_ends_with' ) ) {
+			$this->markTestSkipped( 'str_ends_with() is not available.' );
+		} else {
+			$this->assertSame(
+				$expected,
+				str_ends_with( $string, $needle )
+			);
+		}
+
+	}
+
+	public function str_ends_with_provider() {
```

Comment:
Should providers have DocBlocks?  I don't know...just asking.

Also, I think it would be a good idea to add a couple more test cases to cover:

- string(s) that contains the null character (`\x00`)
- strings with non-ASCII characters, especially multibyte characters (e.g., the trademark symbol `™`, `\x2122`)

The reason is that the PHP manual explicitly says that [str_ends_with()](https://www.php.net/manual/en/function.str-ends-with.php) is "binary-safe" but the PGP manual for [strlen()](https://www.php.net/manual/en/function.strlen.php) does not...and in fact, I didn't think `strlen()` was binary safe...and if `strlen()` was **not** binary-safe the polyfill implementation in this PR probably wouldn't work.

I also wonder whether including a test case for strings in a right-to-left (RTL) language, such as Arabic or Hebrew.  I don't know a lot about RTL languages, so just wondering what the PHP 8 implementation does with such strings.

Reply:
My guess is, RTL language test cases aren't necessary...because I doubt the PHP 8 functions are "LTR/RLT" aware...but I just wanted to raise the issue.